pyiceberg/manifest.py

```diff
@@ -289,10 +286,7 @@ def partition_field_to_data_file_partition_field(partition_field_type: IcebergTy
 
 
 @partition_field_to_data_file_partition_field.register(LongType)
-@partition_field_to_data_file_partition_field.register(DateType)
```

Comment:
This single-dispatch is there only for the `TimeType` it seems. Probably we should we should also convert those into a native type.

Reply:
fixed in the commit [82dd3ad](https://github.com/apache/iceberg-python/pull/555/commits/82dd3adff485c732d24158103ef43cd2e14ded8e)

Reply:
Beautiful, thanks 👍